[Moo Jun Wei] iP

junweimoo opened the pull request:
## Duke
> “Life is not a list of tasks but moments of experiences.” 
#### Features
- Add todos, events, and deadlines
- Delete tasks
- List all tasks
- Search for tasks
- Mark and unmark tasks
#### Downloads
Duke is **FREE** to download and use! ([releases](https://github.com/junweimoo/ip/releases))
#### Quick Start
1. Run the app
2. Add a new todo: `todo buy groceries`
3. List your tasks: `list`
4. Mark your task as done: `mark 1`
#### Implementation
Here's the _main_ method:
```java
public static void main(String[] args) {
    Application.launch(Duke.class, args);
}
```
#### To be Completed
- [ ] Better GUI 😃 
- [ ] Give Duke a personality 😎 
- [ ] More unit tests 👍

junweimoo committed:
Add Greet, Echo, and Exit functionality

junweimoo committed:
Add ability to store and retrieve inputs

junweimoo committed:
Add ability mark and unmark tasks

junweimoo committed:
Support adding different types of tasks

junweimoo committed:
Add automated text UI testing

junweimoo committed:
Handle input errors

junweimoo committed:
Add ability to delete tasks

junweimoo committed:
Refactor code for better readability

junweimoo committed:
Add ability to persist data using a save file

junweimoo committed:
Store date as a LocalDate object instead of a String

junweimoo committed:
Merge branch 'branch-Level-8'

junweimoo committed:
Fix printing of task date

alvinjiang1 commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+    @Override
+    public String toDataString() {
+        return String.format("[D] | %d | %s | %s",
+                isMarked() ? 1 : 0,
```
Perhaps you could look into replacing the 1 and 0 in the conditional statement with a named constant? It could be considered a "magic number".

alvinjiang1 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,218 @@
+
+    public void start() {
+        Scanner sc = new Scanner(System.in);
+        System.out.println(
+                "Hello! I'm Duke\n" +
+                "What can I do for you?"
+        );
```
Could perhaps break up the line before the operator, in accordance with the coding standards?

alvinjiang1 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,218 @@
+        loop(sc);
+    }
+
+    public void loop(Scanner sc) {
```
This method looks long and complex, could think of abstracting out key components into different methods?

alvinjiang1 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,218 @@
+import java.util.List;
+import java.util.Scanner;
+
 public class Duke {
```
It would be great if javadocs are included :))

alvinjiang1 reviewed (commented):
Other than certain long methods and some small issues regarding coding standards, I think the code is fine. Keep up the good work! You are doing great :))

junweimoo committed:
Refactor code to be more object-oriented

junweimoo committed:
Put all classes into the duke package

junweimoo committed:
Add gradle

junweimoo committed:
Add JUnit tests for DataParser and Parser

junweimoo committed:
Refactor parsing code into separate classes

junweimoo committed:
Handle more invalid commands

junweimoo committed:
Add find command to search the list of tasks

junweimoo committed:
Add JavaDoc comments

junweimoo committed:
Modify code to comply with the coding standard.

junweimoo committed:
Merge branch 'branch-A-JavaDoc'

junweimoo committed:
Merge branch 'branch-A-CodingStandard'

junweimoo committed:
Add JavaFX GUI

junweimoo committed:
Improve code quality in InputParser

Previously, the parse() method is lengthy. The body is divided into two main
responsibilities - separating the input into the command keyword and the body,
and then parsing them to return a Command. Code blocks are extracted into two
private methods to improve the readability of the code and adhere to a single level\
of abstraction within parse().

junweimoo committed:
Merge pull request #2 from junweimoo/branch-A-CodeQuality

Improve code quality in InputParser

junweimoo committed:
Refactor parseCommand() in InputParser

junweimoo committed:
Add assertions to TaskList and Storage

junweimoo committed:
Merge pull request #3 from junweimoo/branch-A-Assertions

Add assertions to TaskList and Storage

junweimoo committed:
Add ability to undo commands

Performing consecutive undos will undo commands one-by-one in order of the
history of executed commands. Undo itself is not included in the history of executed
commands.

junweimoo committed:
Merge pull request #4 from junweimoo/branch-BCD-Extension

Add ability to undo commands

junweimoo committed:
Remove unneeded Ui class

junweimoo committed:
Improve the GUI

Lan-Jingbo commented on src/main/java/duke/utils/InputParser.java:
```diff
@@ -23,23 +23,27 @@ public Command parse(String input, TaskList tasks, Storage storage, Ui ui) {
+                return new AddTaskCommand(storage, ui, tasks, TaskType.TODO, body);
+            case "deadline":
+                return new AddTaskCommand(storage, ui, tasks, TaskType.DEADLINE, body);
+            case "event":
```
I think this change(from if to swtich) is good!

junweimoo committed:
Fix truncation of longer lists by dialog boxes

junweimoo committed:
Add more JUnit tests

junweimoo committed:
Add user guide and screenshot

junweimoo committed:
Refactor methods in Storage for better readability

junweimoo committed:
Add JUnit Tests for Storage and TaskParser

junweimoo committed:
Update user guide

junweimoo committed:
Add JavaDoc comments for public methods

junweimoo committed:
Improve the appearance of the GUI

junweimoo committed:
Update UI screenshot

junweimoo committed:
Improve handling of empty and invalid commands

junweimoo committed:
Add more JUnit tests for UndoCommand and InputParser

junweimoo committed:
Remove unused imports

junweimoo committed:
Fix undo delete task to add task back to correct position

junweimoo committed:
Add a welcome message

junweimoo committed:
Fix indentation of switch statement

junweimoo committed:
Add JavaDoc comments for constructors

junweimoo committed:
Add JavaDoc for classes

junweimoo committed:
Update user guide

junweimoo committed:
Update user guide

junweimoo committed:
Update user guide

junweimoo committed:
Add JavaDoc comments

junweimoo committed:
Update user guide

junweimoo committed:
Fix code style

junweimoo committed:
Make dialog box resize to fit text width

junweimoo committed:
Update screenshot
